frontend/README.md

```diff
@@ -0,0 +1,389 @@
+- Includes a tagging system for cache, so you can decide when to refetch certain queries.
+
+**Cons:**
+
+- It doesn't include a custom state management.
+- It still uses hooks a primary API.
+
+### Decision
+
+Having worked on big enterprise level React apps and at the same time on smaller code bases, there are some compromises that you need to do based on the team, state of the business and overall development experience.
+
+I believe that having overkill startegies when is not necessary can be as harmful as not having them when its necessary as it can slow the development process by adding extra overhead or quirks just to support something that can be simply done by just using React.
+
+In this case the best combination for a small app like this, I believe that a combination of React's core state management system like hooks, providers, context, etc. Combined with a simple but powerful I/O tool like `react-query` should be the best fit.
+
+As `react-query` allows you control the actual API request, you can create base gateways and custom responses (models) to use across the app, and the `providers` should be able to spread the information by custom hooks.
+
+### State Data Structure
+
+As `react-query` will do the heavy lifting to keep the image state and information in memory. The thing that we need to keep track will be the list of `in-progress` uploads. In this case we'll be using a native React [reducer](https://beta.reactjs.org/reference/react/useReducer), where we are going to store a key value map for the uploads, it should look as the following:
+
+```json
+{
+ "upload-id":  UploadModel,
+ "upload-id-2":  UploadModel,
+}
+```
+
+And accessing the store to write should be done by dispatching changes to `add, delete and update`.
+
+## Design-System
+
+To ensure component reusability and avoid recreating the same components over and over, creating an set of initial `design-system` component which should provide the basic styles for commonly used controls like:
+
+1. Buttons
+2. Text
+3. Modals
+4. Colors
+
+### Style-Components
+
+The styled-component library becomes very handy when it comes to create extendible components based on a specific set of styles, it contains a `ThemeProvider` tha can be used to store the overall styling rules, like spacing, colors, fonts, etc.
+
+### Storybook
+
+The Storybook library is another great tool that can be used to create generic components while visualizing them before wiring them up with the rest of the application.
+
+## Application Entities
+
+- [Components](entities/Components.md)
+- [Models](entities/Models.md)
+- [Services](entities/Services.md)
+- [Gateways](entities/Gateways.md)
+- [Pages](entities/Pages.md)
+- [Types](entities/Types.md)
+
+## Rendering the Home Page
+
+Acceptance Criteria:
+
+AC1:
+As a user landing on the main page
+I should be able to see the loading state with skeleton items
+
+AC2:
+As a user landing on the main page
+And the requests are finish loading
```

Comment:
```suggestion
And the requests are finished loading
```
Because they don't show up as multi-line on the finished document the capitalization in the middle of the sentence looks weird. I came to correct it when I realized you wrote it as multi-line here, so I won't make those corrections because I think in conjunction with dealing with the markdown formatting for multi-line you'll resolve this issue. If they're multi-line this capitalization looks good. 👍